src/reactor.cc

```diff
@@ -0,0 +1,333 @@
+      ->AddVal("Time", context()->time())
+      ->AddVal("Event", name)
+      ->AddVal("Value", val)
+      ->Record();
+}
+
+void Reactor::RecordStatus(std::string status, double power) {
+  context()
+      ->NewDatum("ReactorStatus")
+      ->AddVal("AgentId", id())
+      ->AddVal("Time", context()->time())
+      ->AddVal("Status", status)
+      ->AddVal("Power", power)
+      ->Record();
+}
+
+void Reactor::RecordInventories(double storage, double excess, double sequestered, 
+                            double blanket, double blanket_excess, double helium) {
+  context()
+      ->NewDatum("ReactorInventories")
+      ->AddVal("AgentId", id())
+      ->AddVal("Time", context()->time())
+      ->AddVal("TritiumStorage", storage)
+      ->AddVal("TritiumExcess", excess)
+      ->AddVal("TritiumSequestered", sequestered)
+      ->AddVal("LithiumBlanket", blanket)
+      ->AddVal("BlanketExcess", blanket_excess)
+      ->AddVal("HeliumStorage", helium)
+      ->Record();
+}
+
+void Reactor::DepleteBlanket(double bred_tritium_moles) {
+  cyclus::Material::Ptr blanket_mat = blanket.Pop();
+  cyclus::toolkit::MatQuery b(blanket_mat);
+  cyclus::CompMap depleted_comp;
+
+
+  double bred_He4 = bred_tritium_moles;
+  double blanket_tritium = b.moles(tritium_id);
+  double blanket_He4 = b.moles(He4_id);
+  double remaining_Li6 = b.moles(Li6_id) - Li6_contribution 
+                          * bred_tritium_moles;
+  double remaining_Li7 = b.moles(Li7_id) - Li7_contribution 
+                          * bred_tritium_moles;
+
+  // This is ALMOST the right behavior, not "scraping the bottom of the barrel
+  if ((remaining_Li6 > 0) && (remaining_Li7 > 0)) {
+    depleted_comp = {{Li7_id, remaining_Li7},
+                     {Li6_id, remaining_Li6},
+                     {tritium_id, blanket_tritium + bred_tritium_moles},
+                     {He4_id, blanket_He4 + bred_He4}};
+
+    cyclus::compmath::Normalize(&depleted_comp, 1);
+
+    // Because there's a mass difference between T+He and Li we need a new mass
+    double new_mass = ((remaining_Li7) * Li7_atomic_mass
+                    + (remaining_Li6) * Li6_atomic_mass
+                    + (blanket_tritium + bred_tritium_moles) * tritium_atomic_mass
+                    + (blanket_He4 + bred_He4) * He4_atomic_mass)*avagadros_number;
+
+    blanket_mat = cyclus::Material::Create(this, new_mass,
+                    cyclus::Composition::CreateFromAtom(depleted_comp));
+
+    RecordOperationalInfo("Blanket Depletion",
+                          "Tritium bred at perscribed rate");
+  } else {
+    RecordOperationalInfo(
+        "Breeding Error",
+        "Blanket composition lacks sufficient lithium to continue "
+        "breeding at perscribed rate");
+  }
+  blanket.Push(blanket_mat);
+}
+
+cyclus::Material::Ptr Reactor::BreedTritium(double atoms_burned, double TBR) {
+  DepleteBlanket(atoms_burned/avagadros_number * TBR);
+  cyclus::Material::Ptr mat = blanket.Pop();
```

Comment:
The last thing that `DepleteBlanket()` does is push a material onto the buffer and then the first thing you do here is pop it off the buffer again.  Maybe `DepleteBlanket()` should just return the material ptr?